boefjes/boefjes/app.py

```diff
@@ -175,13 +186,14 @@ def exit(self, signum: int | None = None) -> None:
                 logger.info("Received %s, exiting", signal.Signals(signum).name)
 
             if not self.task_queue.empty():
-                items: list[Task] = [self.task_queue.get() for _ in range(self.task_queue.qsize())]
+                items: list[list[Task]] = [self.task_queue.get() for _ in range(self.task_queue.qsize())]
 
-                for p_item in items:
-                    try:
-                        self.scheduler_client.push_item(p_item)
-                    except HTTPError:
-                        logger.exception("Rescheduling task failed[id=%s]", p_item.id)
+                for p_items in items:
+                    for p_item in p_items:
```

Comment:
You could use https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable to flatten the lists to skip the second loop, or leave as is